iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -166,13 +168,89 @@ void PortManager::handleReceiverPorts()
     }
 }
 
-void PortManager::handleInterfaces()
+void PortManager::handlePublisherPorts() noexcept
+{
+    // get the changes of publisher port offer state
+    for (auto publisherPortData : m_portPool->getPublisherPortDataList())
+    {
+        PublisherPortRouDiType publisherPort(publisherPortData);
+        auto maybeCaproMessage = publisherPort.getCaProMessage();
+        if (maybeCaproMessage.has_value())
+        {
+            auto& caproMessage = maybeCaproMessage.value();
+
+            m_portIntrospection.reportMessage(caproMessage);
+
+            if ((capro::CaproMessageType::OFFER == caproMessage.m_type)
+                || (capro::CaproMessageType::STOP_OFFER == caproMessage.m_type))
+            {
+                addEntryToServiceRegistry(caproMessage.m_serviceDescription.getServiceIDString(),
+                                          caproMessage.m_serviceDescription.getInstanceIDString());
+
+                sendToAllMatchingSubscriberPorts(caproMessage, publisherPort);
+            }
+            else
+            {
+                // protocol error
+                errorHandler(Error::kPORT_MANAGER__HANDLE_PUBLISHER_PORTS_INVALID_CAPRO_MESSAGE,
+                             nullptr,
+                             iox::ErrorLevel::MODERATE);
+            }
+
+            // forward to interfaces
+            sendToAllMatchingInterfacePorts(caproMessage);
+        }
+        else
+        {
+            errorHandler(
+                Error::kPORT_MANAGER__HANDLE_PUBLISHER_PORTS_NOT_A_CAPRO_MESSAGE, nullptr, iox::ErrorLevel::MODERATE);
+        }
+
+        // check if we have to destroy this sender port
+        if (publisherPort.toBeDestroyed())
+        {
+            destroyPublisherPort(publisherPortData);
+        }
+    }
+}
+
+void PortManager::handleSubscriberPorts() noexcept
+{
+    // get requests for change of subscription state of subscribers
+    for (auto subscriberPortData : m_portPool->getSubscriberPortDataList())
+    {
+        SubscriberPortProducerType subscriberPort(subscriberPortData);
+        auto returnedCaproMessage = subscriberPort.getCaProMessage();
+        if (returnedCaproMessage.has_value())
+        {
+            auto& caproMessage = returnedCaproMessage.value();
+
+            m_portIntrospection.reportMessage(caproMessage);
+
+            if (!sendToAllMatchingPublisherPorts(caproMessage, subscriberPort))
+            {
+                LogDebug() << "capro::SUB/UNSUB, no matching sender!!";
+                capro::CaproMessage nackMessage(capro::CaproMessageType::NACK,
+                                                subscriberPort.getCaProServiceDescription());
+                subscriberPort.dispatchCaProMessage(nackMessage);
+            }
+        }
+
```

Comment:
shouldn't there be an else branch like with `PortManager::handlePublisherPorts`

Reply:
What is it supposed to do in case of the else branch? It used to do nothing. Should it dispatch an `ok` message?

Reply:
I would assume it's an error, like with `PortManager::handlePublisherPorts`. @budrus, what do you think?
Keep in mind, this was quite old code where the error handling was not optimal